### PR TITLE
Feat (Advanced Settings): Make new "Appearance" category

### PR DESCRIPTION
**Pull request by joshuarrrr**
### Description

And remove "Accessibility".
Give category panels IDs so they can be deep linked

Copy and other settings updates will follow in separate PRs.

### Issues Resolved

Partial implementation of https://github.com/opensearch-project/OpenSearch-Dashboards/issues/4330
Unblocks #4676 

## Screenshot

Note that advanced settings are automatically ordered alphabetically by the setting name.

https://github.com/opensearch-project/OpenSearch-Dashboards/assets/1679762/f81691fb-fb77-4017-b701-c5f88cd04e7e

### Check List

- [ ] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
  - [ ] `yarn test:ftr`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [ ] Commits are signed per the DCO using --signoff

### CODE REVIEW

**Comment by joshuarrrr** [comment]
@KrooshalUX This is ready for your review. Note that it's not everything from https://github.com/opensearch-project/OpenSearch-Dashboards/issues/4330, but it does have everything moved to the new section, which is deep linkable. Given this minimal implementation, should I go ahead and also update the copy to match the mock (except for the browser settings, which won't be supported in 2.10).

@mnkugler The issue mentions that we want the theme description text to link to a forum post. Can you provide that forum post URL?

**Comment by KrooshalUX** [comment]
👍  this is great incremental progress towards the north star of appearance within advanced settings (#4330 )!

**Comment by kavilla** [comment]
Not sure if this a current issue but will we move forward with the other plugins to add to this category?

**Comment by ashwin-pc** [comment]
@joshuarrrr Some of the unit tests are failing here, can you look at them?

**Comment by joshuarrrr** [comment]
> Not sure if this a current issue but will we move forward with the other plugins to add to this category?

Are you asking whether plugins can add additional settings to this Appearance category? If so, the answer is yes, which is consistent with how the other categories work.

File: src/core/server/ui_settings/settings/theme.ts [hunk 1]
```diff
@@ -53,9 +54,11 @@ export const getThemeSettings = (): Record<string, UiSettingsParams> => {
       type: 'select',
       options: ['v7', 'Next (preview)'],
       description: i18n.translate('core.ui_settings.params.themeVersionText', {
-        defaultMessage: `Switch between the theme used for the current and next version of OpenSearch Dashboards, A page refresh is required for the setting to be applied.`,
+        defaultMessage: `<p>Switch between the theme used for the current and next version of OpenSearch Dashboards. A page refresh is required for the setting to be applied.</p><p><a href="{href}">{linkText}</a></p>`,
```
Comment (AMoo-Miki) [inline src/core/server/ui_settings/settings/theme.ts:57]:
```suggestion
        defaultMessage: `<p>Switch between the themes used for the current and next version of OpenSearch Dashboards. A page refresh is required for the setting to be applied.</p><p><a href="{href}">{linkText}</a></p>`,
```

**Review by AMoo-Miki — APPROVED** [review]

**Review by ashwin-pc — APPROVED** [review]

File: src/core/server/ui_settings/settings/theme.ts [hunk 2]
```diff
@@ -53,9 +54,11 @@ export const getThemeSettings = (): Record<string, UiSettingsParams> => {
...
+        values: { href: 'https://forum.opensearch.org/', linkText: 'Theme feedback' },
```
[the remark this answers is not in the document]
Comment (joshuarrrr) [reply src/core/server/ui_settings/settings/theme.ts]: href updated

File: src/core/server/ui_settings/settings/theme.ts (hunk 1 above)
Comment (KrooshalUX) [reply src/core/server/ui_settings/settings/theme.ts:57]: @joshuarrrr  work with documentation for copy approval.

**Review by ashwin-pc — DISMISSED** [review]

File: src/core/server/ui_settings/settings/theme.ts (hunk 1 above)
Comment (vagimeli) [reply src/core/server/ui_settings/settings/theme.ts:57]: @joshuarrrr @AMoo-Miki The above suggested changes are correct.

**Review by kavilla — APPROVED** [review]

**Review by ashwin-pc — APPROVED** [review]

**Pull request merged**